### PR TITLE
[CI] GitHub Actions Cocoapods Caching

### PR DESCRIPTION
**Pull request by anayini**
## Changes in this pull request

This introduces caching in GitHub Actions so that we don't re-resolve our Pods each time.

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/plangrid/ReactiveLists/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by anayini** [comment]
This now seems to cache the `Pods/` folder which is a pretty minor optimization (might save us something like ~10 seconds in a simple project like this with few dependencies).  It seems like a lot of time is being taken in cloning the master Specs repo.  I'll follow up separately to see if I can find a good solution to that

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name: iOS Test
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        destination: ["OS=13.2,name=iPhone 11"]
+    steps:
+      - uses: actions/checkout@v1
+      - name: Bundle Install
+        run: bundle install
+      - uses: actions/cache@v1
```
Comment (ronaldsmartin) [inline .github/workflows/ci.yml]: This is the relevant change, right? Any docs you can point me to?

**Comment by ronaldsmartin** [comment]
Also, we can probably shave down the Pods repo cloning time by [switching to the Cocoapods CDN](https://blog.cocoapods.org/CocoaPods-1.7.2/), like we did in the main iOS repo, if you'd like to try that.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (anayini) [reply .github/workflows/ci.yml]: Yup.  You can see more info here: https://github.com/actions/cache

Its a marketplace action so I was able to just `uses` it.  There are examples in their docs for how to use it for CocoaPods projects so I lifted the configuration from [there](https://github.com/actions/cache/blob/master/examples.md#swift-objective-c---cocoapods).

**Comment by anayini** [comment]
Oh cool (re: CocoaPods CDN).  I tried to use the same caching code to see if I could cache `~/.cocoapods`, but the Specs repo is bigger than the cache limit lol

**Comment by jessesquires** [comment]
nice 😎

**Pull request merged**